-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c11f12/38457/summary.html
**COMMIT**: 1053e8cbf1877d8f1f7a7895b1ca0e010150ecf8
**CMSSW**: CMSSW_14_1_X_2024-03-27-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw-config/106/38457/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @artlbv cms-sw/cmssw#44510

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c11f12/38457/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-c11f12/38457/git-merge-result


## Build

I found compilation error when building: 

<pre>>> Cuda Device Link tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaInterface/test/alpakaTestAtomicPairCounterCudaAsync/alpakaTestAtomicPairCounterCudaAsync_cudadlink.o 
>> Building alpaka/cuda binary alpakaTestAtomicPairCounterCudaAsync
Copying tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaInterface/test/alpakaTestAtomicPairCounterCudaAsync/alpakaTestAtomicPairCounterCudaAsync to productstore area:
>> Building alpaka/rocm binary alpakaTestAtomicPairCounterROCmAsync
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/testBuildDir/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: cannot find tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaInterface/test/alpakaTestAtomicPairCounterROCmAsync/alpaka/testAtomicPairCounter.dev.cc.o: No such file or directory
collect2: error: ld returned 1 exit status
>> Deleted: tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaInterface/test/alpakaTestAtomicPairCounterROCmAsync/alpakaTestAtomicPairCounterROCmAsync
gmake: *** [tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaInterface/test/alpakaTestAtomicPairCounterROCmAsync/alpakaTestAtomicPairCounterROCmAsync] Error 1
>> Compiling alpaka/serial src/HeterogeneousCore/AlpakaInterface/test/alpaka/testAtomicPairCounter.dev.cc
>> Building alpaka/serial binary alpakaTestAtomicPairCounterSerialSync
Copying tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaInterface/test/alpakaTestAtomicPairCounterSerialSync/alpakaTestAtomicPairCounterSerialSync to productstore area:
</pre>

--- a/SCRAM/Plugins/BuildRules.py
+++ b/SCRAM/Plugins/BuildRules.py
@@ -1418,10 +1418,11 @@ $(COMMON_WORKINGDIR)/cache/project_links: FORCE_TARGET
                      "{0}_files := $(patsubst {2}/%,%,$(wildcard $(foreach dir,{2},"
                      "$(foreach ext,$(SRC_FILES_SUFFIXES),$(dir)/*.$(ext)))))\n".
                      format(safename, parent, path, bend))
-            if bend=="rocm":
-                self.check_rocm_files("alpaka_device")
             self.pushstash()
             self.set('safename',safename)
+            if bend=="rocm":
+                self.set("check_rocm_files",False)
+                self.check_rocm_files("alpaka_device")
             self.set('use_private', 'alpaka-%s %s' % (bend, self.core.get_flag_value("USE_ALPAKA_" + bend.upper())))
             self.set("classes_file", "classes_%s" % bend)
             self.set("classes_file_type", "ALPAKA_%s_LCG" % bend.upper())
@@ -1643,7 +1644,9 @@ $(COMMON_WORKINGDIR)/cache/project_links: FORCE_TARGET
                     fh.write("%s_CLASS := %s\n" % (safename, ptype))
                     fh.write("%s_PRODUCT_TYPE:=alpaka/%s\n" % (safename, bend))
                     fh.write("%s_files := $(%s_files)\n" % (safename, psafename))
-                    if bend in ["rocm"]: self.check_rocm_files("alpaka_device")
+                    if bend in ["rocm"]:
+                        self.set("check_rocm_files",False)
+                        self.check_rocm_files("alpaka_device")
                     self.dumpBuildFileData(lib, False)
                     self.popstash()
                 self.set("alpaka_names"," ".join(alpaka_names))

